### PR TITLE
Fix/misc

### PR DESCRIPTION
**Pull request by patrickvacek**
Things found while doing other things.

### CODE REVIEW

File: src/libaktualizr/primary/sotauptaneclient.cc [hunk 1]
```diff
@@ -628,7 +628,7 @@ std::pair<bool, Uptane::Target> SotaUptaneClient::downloadImage(Uptane::Target t
     }
   }
   if (!success) {
-    LOG_ERROR << "Download unsuccessful after " << tries << " attempts.";
+    LOG_ERROR << "Download unsuccessful after " << tries - 1 << " attempts.";
```
Comment (eu-smirnov) [inline src/libaktualizr/primary/sotauptaneclient.cc]: How about `for (; tries < max_tries; tries++)` ?

File: src/aktualizr_repo/director_repo.cc [hunk 2]
```diff
@@ -57,11 +57,21 @@ void DirectorRepo::signTargets() {
...
+
+  Json::Value targets_current = Utils::parseJSONFile(current);
   Json::Value targets_unsigned;
   targets_unsigned = Utils::parseJSONFile(staging);
-  targets_unsigned["targets"].clear();
+  targets_unsigned["_type"] = "Targets";
+  targets_unsigned["expires"] = expiration_time_;
+  targets_unsigned["version"] = (targets_current["signed"]["version"].asUInt()) + 1;
+  targets_unsigned["targets"] = Json::objectValue;
+  if (repo_type_ == Uptane::RepositoryType::Director() && correlation_id_ != "") {
```
Comment (mike-sul) [inline src/aktualizr_repo/director_repo.cc]: Is this a fix for the super bug ?

File: src/libaktualizr/primary/sotauptaneclient.cc (hunk 1 above)
Comment (patrickvacek) [reply src/libaktualizr/primary/sotauptaneclient.cc]: Done.

File: src/aktualizr_repo/director_repo.cc (hunk 2 above)
Comment (patrickvacek) [reply src/aktualizr_repo/director_repo.cc]: The super bug? The fix here is just that the generated metadata was incorrect. It had the wrong meaning of "empty".

File: src/libaktualizr/primary/sotauptaneclient.cc [hunk 3]
```diff
@@ -607,22 +607,22 @@ std::pair<bool, Uptane::Target> SotaUptaneClient::downloadImage(Uptane::Target t
...
+  const int max_tries = 3;
+  int tries = 0;
+  std::chrono::milliseconds wait(500);
+
+  for (; tries < max_tries; tries++) {
     success = package_manager_->fetchTarget(target, *uptane_fetcher, keys, prog_cb, token);
     if (success) {
       break;
-    } else if (tries < max_tries) {
+    } else if (tries < max_tries - 1) {
```
Comment (eu-smirnov) [inline src/libaktualizr/primary/sotauptaneclient.cc]: So we won't sleep in the last iteration (i = 2)? My guess would be that we don't need an `else if` here.

Comment (patrickvacek) [reply src/libaktualizr/primary/sotauptaneclient.cc]: Right. Why bother sleeping if we've tried three times and failed? No need to sleep before moving on to the next thing.

Comment (eu-smirnov) [reply src/libaktualizr/primary/sotauptaneclient.cc]: Sure. Somehow I always get confused by this simple loop.

Comment (patrickvacek) [reply src/libaktualizr/primary/sotauptaneclient.cc]: Me too. This is at least the third time we've tried to get it right despite it being so deceptively simple.

**Pull request merged**